File: accenture/Naukri360/_16.cpp
#include <iostream>
#include <unordered_map>
#include <algorithm>
#include <vector>
using namespace std;

int main()
{
    int n;
    cin >> n;
    vector<int> arr(n);
    unordered_map<int, int> freq;
    for (int i = 0; i < n; i++)
    {
        cin >> arr[i];
        freq[arr[i]]++;
    }
    sort(arr.begin(),arr.end(),[&](int a,int b){
        if(freq[a] == freq[b]){
            return a < b;
        }
        return freq[a] > freq[b];
    });
    std::cout << arr[0];

    return 0;
}

File: accenture/Naukri360/_16.py
n = int(input())
arr = [0] * n
count = {}
for i in range(n):
    arr[i] = int(input())
    count[arr[i]] = 0
for i in range(n):
    count[arr[i]] +=1

arr.sort(key = lambda x:(-count[x],x))
print(arr[0])